src/messages.lcm

```diff
@@ -1,25 +1,99 @@
+// -*-C-*-
 package robot;
 
-struct dwm_position_t
+// Found on channel IO_POSITION
+struct io_position_t
 {
         int64_t timestamp; // in milliseconds
+
+        // Coordinates in meters
         float x;
         float y;
         float z;
+
+        // Quality indicator. Not linear; treat as some enum ranging
+        // from BAD to GOOD. Higher value is better.
         int8_t q;
 }
 
-struct dwm_acceleration_t
+// Found on channel IO_ACCELERATION
+struct io_acceleration_t
 {
         int64_t timestamp; // in milliseconds
+
+        // Coordinates in ms^{-2}
         float x;
         float y;
         float z;
 }
 
-struct vector2d_t
+struct brickpi_t
+{
+        const int8_t PLATFORM = 1, ARM = 2;
+}
+struct port_t
+{
+        const int8_t PORT_1 = 1,
+                PORT_2 = 2,
+                PORT_3 = 3,
+                PORT_4 = 4,
+                PORT_A = 5,
+                PORT_B = 6,
+                PORT_C = 7,
+                PORT_D = 8;
+}
+// Found on channel IO_ENCODER
+struct io_encoder_t
+{
+        brickpi_t brickpi;
+        port_t port;
+        int64_t dt;
+}
```

Comment:
Here we also need a counter that holds how many ticks were measured during `dt`.